lib/wickedbase.pm

```diff
@@ -117,6 +119,104 @@ sub assert_wicked_state {
     $self->ping_with_timeout(ip => $args{ping_ip}) if $args{ping_ip};
 }
 
+sub valgrind_get_services {
+    my $self = shift;
+    my $val = get_var('WICKED_VALGRIND');
+
+    return if (!defined($val) || !$val);
+    my @all = qw(wickedd-auto4
+      wickedd-dhcp6
+      wickedd
+      wickedd-dhcp4
+      wickedd-nanny
+      wicked);
+    my @tmp;
+    my @enable;
+    if (index($val, ",")) {
+        @tmp = split(/,/, $val);
+    } else {
+        @tmp = $val;
+    }
+    foreach my $v (@tmp) {
+        $v =~ s/\.service$//;
+        if (grep { /^$v$/ } @all) {
+            push @enable, $v;
+        } elsif ($v =~ /^1|all$/i) {
+            push @enable, @all;
+        } else {
+            record_info('WARNING', "Unknown value for WICKED_VALGRIND $v", result => 'softfail');
+        }
+    }
+    @enable = uniq @enable;
+
+    return @enable;
+}
+
+sub valgrind_cmd {
+    my ($self, $service) = @_;
+
+    my $valgrind_cmd = get_var('WICKED_VALGRIND_CMD', '/usr/bin/valgrind --tool=memcheck --leak-check=yes');
+    if ($service) {
+        my $logfile = "/var/log/valgrind_$service.log";
+        $valgrind_cmd = "$valgrind_cmd --log-file=$logfile";
+    }
+    return $valgrind_cmd;
+}
+
+sub valgrind_enable {
+    my $self = shift;
+    my $valgrind_cmd = get_var('WICKED_VALGRIND_CMD', '/usr/bin/valgrind --tool=memcheck --leak-check=yes');
+
+    my @services = $self->valgrind_get_services();
+    return 0 if (!@services);
+
+    record_info("valgrind enable", "services: @services\ncommand: $valgrind_cmd");
+
+    foreach my $service (@services) {
+        my $service_file = "/etc/systemd/system/$service.service";
+
+        assert_script_run("systemctl cat $service > $service_file");
```

Comment:
I don't understand this line , isn't you output'ing  into service file what was already there ?

Reply:
Exactly. But it is a different directory. This is the directory where "custom" changes can be made. The RPM put it in a system directory like `/lib/systemd/system/` and I made my changes in `/etc/systemd/system/`  , see: https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Unit%20File%20Load%20Path